### ISSUE TITLE
V3-32 and V3-8 update freq 4 doesnt converge but GPU can

### ISSUE THREAD

**Issue by kkissmart** [pytorch/xla#2354]
Please see this issue for detailed logs
They are on exactly the same code and data

https://github.com/pytorch/fairseq/issues/2332

any thought on why this happens?

**Comment by taylanbil** [comment]
Looking at the two logs; it seems like your gpu epoch consists of 2950978 steps whereas tpu consists of 46127. This is off by a factor of 64. Since you're saying exact same data, this is most likely due to a difference in global batch size (`batch size per device` * `update-freq` * `num devices`). So it's a bit apples-to-oranges. For example, when batch size changes, one usually also changes the learning rate. If you wanted to make an exact comparison, I suggest to adjust parameters so it's more apples to apples.

**Comment by taylanbil** [comment]
Also, this issue title mentions v3-32 but there's no info on what's going on with v3-32.

**Comment by kkissmart** [comment]
I only have 2 local gpus , but also tried GPU with update-freq 64 before. the logs were similar (better, since bsz is larger). V3-32 logs are even worse -- got loss got stuck at 1000 step.

**Comment by kkissmart** [comment]
@taylanbil i can share with you my data if you need to reproduce logs on gpu and tpu.

**Comment by taylanbil** [comment]
This was probably a related issue to https://github.com/pytorch/fairseq/issues/2338, we synced and closing this until @kkissmart can repro this again.
